Source/driver/Castro.cpp

```diff
@@ -3551,40 +3573,59 @@ Castro::clean_state(MultiFab& state) {
     // Compute the temperature (note that this will also reset
     // the internal energy for consistency with the total energy).
 
-    computeTemp(state);
+    computeTemp(state, state.nGrow());
 
     return frac_change;
 
 }
 
 
-
 Real
-Castro::clean_state(MultiFab& state, MultiFab& state_old) {
+Castro::clean_state(int is_new, int ng) {
+
+  // this is the "preferred" clean_state interface -- it will work
+  // directly on the StateData.  is_new=0 means the old data is used,
+  // is_new=1 means the new data is used.
 
-    // Enforce a minimum density.
 
 #ifndef AMREX_USE_CUDA
+    MultiFab& state = is_new == 1 ? get_new_data(State_Type) : get_old_data(State_Type);
     Real frac_change = enforce_min_density(state_old, state);
```

Comment:
Is this call supposed to be in here? I would expect it to be in the actual clean_state interface below.

Reply:
I think I put this here only for the frac_change bits.  I need to know which state I am enforcing the minimum density on.

Reply:
oh... I think this is a merge issue.  Let me fix

Reply:
right, the call is not supposed to be there.  Nor do I need the `AMREX_USE_CUDA` logic there, it belongs in the actual `clean_state`.  Fixed now.